src/lib/Microsoft.Health.Fhir.Ingest/Service/EventProcessingMeter.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class EventProcessingMeter : IEventProcessingMeter
+    {
+        public Task<EventStats> CalculateEventStats(EventData[] events)
+        {
+            double ingressSizeBytes = 0;
+            foreach (var e in events)
+            {
+                var body = e.Body.Array.Length;
+
+                foreach (KeyValuePair<string, object> entry in e.Properties)
```

Comment:
Fyi; some nice one-liners to to replace these loops using Linq:

```
ingressSizeBytes = e.Properties.Aggregate(0.0, (current, entry) => current + Encoding.UTF8.GetBytes(entry.Key + entry.Value).Length)
ingressSizeBytes = e.SystemProperties.Aggregate(ingressSizeBytes, (current, entry) => current + Encoding.UTF8.GetBytes(entry.Key + entry.Value).Length)
```

Reply:
Thanks, that is a lot cleaner. will update